api/v1/controllers/generators.js

```diff
@@ -219,10 +219,14 @@ module.exports = {
     .then((_collectors) => {
       // prevent overwrite of reloaded collectors on update
       delete puttableFields.possibleCollectors;
+
+      // mock possibleCollectors on instance so we don't need to reload
+      // again to get the currentCollector
+      instance.possibleCollectors = _collectors;
       return instance.setPossibleCollectors(_collectors);
     })
-    .then(() => instance.reload())
     .then(() => u.updateInstance(instance, puttableFields, toPut))
+    .then(() => instance.reload())
```

Comment:
But you also changed assignToCollector to set the currentCollector directly. Did you do that after you added this here? If so, maybe this is no longer necessary?

Reply:
We discussed offline. Simply moving it above doesn't work. It might be possible to get it working without the mock, but it's working as-is and not worth the extra effort at this point.